### PR TITLE
Feature: show advices for TUF updates

### PR DESCRIPTION
**Pull request by vkhoroz**
In the new API we return some advices how to improve the customer's TUF metadata. This change will show these advices to the user.

Signed-off-by: Volodymyr Khoroz <volodymyr.khoroz@foundries.io>

### CODE REVIEW

**Comment by vkhoroz** [comment]
Some examples:

- without an active transaction:
```
There are no TUF root updates in progress.

Foundries.io recommends these updates to improve your existing TUF metadata:
 - Use Ed25519 keys to improve speed and security (e.g. run the command 'fioctl keys tuf rotate-all-keys --keys=tuf-root-keys.tgz --targets-keys=tuf-targets-keys.tgz --key-type=ed25519')

Please, also consider examples from 'fioctl keys tuf updates --help'.
```

- with the newly created transaction:
```
The following TUF root updates are staged for your factory:
 - User <unknown user (id=admin)> at 2023-02-06T19:27:50Z set change message to: Rotate targets key to Ed25519
 - Changed version from 81 to 82

These updates to your staged TUF root are mandatory before applying it:
 - Stage some changes to your TUF root (e.g. run the command 'fioctl keys tuf updates --help' to see some examples)
 - Sign your staged TUF root updates with you offline root key (e.g. run the command 'fioctl keys tuf updates sign --keys=tuf-root-keys.tgz')

Foundries.io also recommends these updates to improve your TUF metadata:
 - Use Ed25519 key for targets to improve speed and security (e.g. run the command 'fioctl keys tuf updates rotate-offline-key --role=targets --targets-keys=tuf-targets-keys.tgz --key-type=ed25519')

Once your are satisfied with your TUF updates, please, run 'fioctl keys tuf updates apply'.
If you want to cancel staged TUF updates, please, run 'fioctl keys tuf updates cancel'.
```

**Comment by doanac** [comment]
commenting on your comment:
```
These updates to your staged TUF root are mandatory before applying it:
 - Stage some changes to your TUF root (e.g. run the command 'fioctl keys tuf updates --help' to see some examples)
 - Sign your staged TUF root updates with you offline root key (e.g. run the command 'fioctl keys tuf updates sign --keys=tuf-root-keys.tgz')
```
That's confusing "Stage some changes ...". I think they just need to sign the staged updates and its ready?

File: subcommands/keys/tuf_updates_review.go [hunk 1]
```diff
@@ -101,20 +103,50 @@ func doTufUpdatesReview(cmd *cobra.Command, args []string) {
 				}
 			}
 		}
+	} else if updates.Status == client.TufRootUpdatesStatusNone {
+		fmt.Println("There are no TUF root updates in progress.")
+		if len(updates.Advices.Mandatory) > 0 {
+			fmt.Println(`\nFoundries.io identified some problems with your existing TUF metadata.
```
Comment (doanac) [inline subcommands/keys/tuf_updates_review.go:109]: We should probably avoid the topic of branding and just say. "Problems have been identified with your TUF metadata"

File: subcommands/keys/tuf_updates_review.go [hunk 2]
```diff
@@ -101,20 +103,50 @@ func doTufUpdatesReview(cmd *cobra.Command, args []string) {
...
+Please, consider these updates in order to keep your TUF metadata functional and secure:`)
```
Comment (doanac) [inline subcommands/keys/tuf_updates_review.go:110]: "secure" is a bit too subjective. If its not secure - the TUF client/server wouldn't allow it in the first place. I'd just drop the entire sentence - cause its not something "to consider" - you *have* to do it.

File: subcommands/keys/tuf_updates_review.go [hunk 3]
```diff
@@ -101,20 +103,50 @@ func doTufUpdatesReview(cmd *cobra.Command, args []string) {
...
+			for _, advice := range updates.Advices.Mandatory {
+				fmt.Printf(" - %s\n", advice.Message)
+			}
+		}
+		if len(updates.Advices.Recommended) > 0 {
+			fmt.Println("\nFoundries.io recommends these updates to improve your existing TUF metadata:")
```
Comment (doanac) [inline subcommands/keys/tuf_updates_review.go:116]: I'd drop the branding here also and just say something like "The following updates are recommended..."

File: subcommands/keys/tuf_updates_review.go [hunk 4]
```diff
@@ -101,20 +103,50 @@ func doTufUpdatesReview(cmd *cobra.Command, args []string) {
...
+			for _, advice := range updates.Advices.Recommended {
+				fmt.Printf(" - %s\n", advice.Message)
+			}
+		}
+		if len(updates.Advices.Mandatory) > 0 || len(updates.Advices.Recommended) > 0 {
+			fmt.Println("\nPlease, also consider examples from 'fioctl keys tuf updates --help'.")
```
Comment (doanac) [inline subcommands/keys/tuf_updates_review.go:122]: I'm on the fence if this adds much value. We've just told them exactly what they need to do, and now we sort of tell them "BTW- please read the manual".

**Comment by mike-sul** [comment]
When the advices are printed? 
Are they printed on each `fioctl keys tuf` subcommand?
Or we are going to introduce a new command to print the advices?
Or they will be printed only by `fioctl keys tuf updates review` command?.

**Comment by vkhoroz** [comment]
> 



> When the advices are printed?
> Are they printed on each `fioctl keys tuf` subcommand?
> Or we are going to introduce a new command to print the advices?
> Or they will be printed only by `fioctl keys tuf updates review` command?.

This PR changes the `fioctl keys tuf updates review` command output; so I _believe_ they are printed by running it.

File: subcommands/keys/tuf_updates_review.go (hunk 1 above)
Comment (vkhoroz) [reply subcommands/keys/tuf_updates_review.go:109]: Makes sense

File: subcommands/keys/tuf_updates_review.go (hunk 3 above)
Comment (vkhoroz) [reply subcommands/keys/tuf_updates_review.go:116]: Ok

File: subcommands/keys/tuf_updates_review.go (hunk 4 above)
Comment (vkhoroz) [reply subcommands/keys/tuf_updates_review.go:122]: I was on the fence here too; so it doesn't hurt me to remove it.

**Comment by vkhoroz** [comment]
> commenting on your comment:
> 
> ```
> These updates to your staged TUF root are mandatory before applying it:
>  - Stage some changes to your TUF root (e.g. run the command 'fioctl keys tuf updates --help' to see some examples)
>  - Sign your staged TUF root updates with you offline root key (e.g. run the command 'fioctl keys tuf updates sign --keys=tuf-root-keys.tgz')
> ```
> 
> That's confusing "Stage some changes ...". I think they just need to sign the staged updates and its ready?

Existing version of fioctl states the following in this case:
```
There are no effective changes to your TUF root except the version and changelog.
These changes cannot be applied before some of the key rotation commands are performed.
Please, run the 'fioctl keys tuf updates --help' for details.
```
I removed this message from fioctl and show the equivalent server-generated message itself.

It is returned when user made no changes to the TUF root. On practice, I _believe_ it is enough to just re-sign the new root; but that really has no effect, so we explicitly instruct the user to use some key rotation commands. Maybe, I should have just copy-pasted the existing message from fioctl to the server side.

Does it make sense to you, @doanac ?

File: subcommands/keys/tuf_updates_review.go (hunk 1 above)
Comment (vkhoroz) [reply subcommands/keys/tuf_updates_review.go]: We now have no errors returned for this use case from the API (as per server-side review thread).

So this code was removed. I applied the branding part elsewhere.

File: subcommands/keys/tuf_updates_review.go (hunk 2 above)
Comment (vkhoroz) [reply subcommands/keys/tuf_updates_review.go]: We now have no errors returned for this use case from the API (as per server-side review thread).

File: subcommands/keys/tuf_updates_review.go (hunk 3 above)
Comment (vkhoroz) [reply subcommands/keys/tuf_updates_review.go]: Done in force-push

File: subcommands/keys/tuf_updates_review.go (hunk 4 above)
Comment (vkhoroz) [reply subcommands/keys/tuf_updates_review.go]: Removed in force-push.

**Review by doanac — APPROVED** [review]

**Pull request merged**